### DISCUSSION TITLE
Canny preprocessor uncertainty

### DISCUSSION THREAD

**Discussion by IPv6** [Acly/krita-ai-diffusion#888] [Q&A] [answered]
What kind of image should be added for Canny/LineArt controlnets? "Make from layer" generate white image with black lines - but all MistoLine/etc examples (and canny controlnets from A1111, etc) are using black image with white lines. Which case is right one?

**Comment by Drakosha405** [comment, marked as answer]
black on white. it is more familiar and convenient for drawing. Next, the program inverts the image itself to read it correctly.
